tests/src/test/scala/jms4s/JmsClientSpec.scala

```diff
@@ -341,12 +341,14 @@ trait JmsClientSpec extends AsyncFreeSpec with AsyncIOSpec with Jms4sBaseSpec {
     res.use {
       case (producer, consumer, bodies, messages) =>
         for {
-          _                <- messages.toNel.fold(IO.unit)(ms => producer.sendN(messageFactory(ms, topicName1)))
+          messageIds <- messages.toNel.fold(IO.pure(List.empty[Option[String]]))(ms =>
+                         producer.sendN(messageFactory(ms, topicName1)).map(_.toList)
+                       )
           _                <- logger.info(s"Pushed ${messages.size} messages.")
           _                <- logger.info(s"Consumer to Producer started.\nCollecting messages from output queue...")
           received         <- Ref.of[IO, Set[String]](Set())
           receivedMessages <- receiveUntil(consumer, received, nMessages).timeout(timeout) >> received.get
-        } yield assert(receivedMessages == bodies)
+        } yield assert(messages.size == messageIds.size && receivedMessages == bodies)
```

Comment:
right now the `messageIds` can be a List of `None`s so i would also check that inside the list there is something, like:
`messageIds.forAll(_.nonEmpty) `.
On the other hand, if the `sendN` will return a List without the option this check is not needed anymore.